Client/Frontend/Browser/ButtonToast.swift

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+struct ButtonToastUX {
+    static let ToastDismissAfter = 4.0
+    static let ToastPadding = 15.0
+    static let ToastButtonPadding:CGFloat = 10.0
+    static let ToastDelay = 0.9
+    static let ToastButtonBorderRadius:CGFloat = 5
+    static let ToastButtonBorderWidth:CGFloat = 1
+}
+
+class ButtonToast {
+    
+    private var dismissed = false
+    private var completionHandler: (Bool) -> Void = {_ in }
```

Comment:
Nit: Probably better to just use an optional here instead of an empty block and conditionally call it when needed like `completionHandler?()`